lib/puppet/util/character_encoding.rb

```diff
@@ -3,79 +3,72 @@
 module Puppet::Util::CharacterEncoding
   class << self
     # Warning! This is a destructive method - the string supplied is modified!
+    # Given a string, attempts to convert the string to UTF-8. Conversion uses
+    # encode! - the string's internal byte representation is modifed to UTF-8.
+    #
+    # This method is intended for situations where we generally trust that the
+    # string's bytes are a faithful representation of the current encoding
+    # associated with it, and can use it as a starting point for transcoding
+    # (conversion) to UTF-8.
+    #
     # @api public
-    # @param [String] string a string to transcode / force_encode to utf-8
-    # @return [String] string if already utf-8, OR
-    #   the same string with external encoding set to utf-8 if bytes are valid utf-8 OR
-    #   the same string transcoded to utf-8 OR
-    #   nil upon a failure to legitimately set external encoding or transcode string
+    # @param [String] string a string to transcode
+    # @return [nil] (string is modified in place)
     def convert_to_utf_8!(string)
```

Comment:
We don't need to address it here, but it seems to me that `convert_to_utf_8` should return a copy of the initial string rather than mutating it...